integrations/tensorflow/compiler/dialect/tf_strings/conversion/convert_tf_to_tf_strings.cc

```diff
@@ -36,47 +36,22 @@ namespace iree_compiler {
 
 namespace {
 
-struct AsStringOpLowering : public OpRewritePattern<TF::AsStringOp> {
-  using OpRewritePattern<TF::AsStringOp>::OpRewritePattern;
-
-  LogicalResult matchAndRewrite(TF::AsStringOp op,
-                                PatternRewriter &rewriter) const override {
-    Value replacement = op.input();
-    replacement =
-        rewriter.create<TFStrings::ToStringOp>(op.getLoc(), replacement);
-    rewriter.replaceOp(op, replacement);
-    return success();
-  }
-};
-
-struct PrintOpLowering : public OpRewritePattern<TF::PrintV2Op> {
-  using OpRewritePattern<TF::PrintV2Op>::OpRewritePattern;
-
-  LogicalResult matchAndRewrite(TF::PrintV2Op op,
-                                PatternRewriter &rewriter) const override {
-    auto input = op.input();
-    rewriter.create<TFStrings::PrintOp>(op.getLoc(), input);
-    rewriter.eraseOp(op);
-    return success();
-  }
-};
+#include "integrations/tensorflow/compiler/dialect/tf_strings/conversion/convert_tf_to_tf_strings.inc"
 
 class StringTypeConverter : public TypeConverter {
  public:
   StringTypeConverter() {
-    addConversion([](ShapedType type) {
-      auto elementType = TFStrings::StringType::get(type.getContext());
-      return RankedTensorType::get(type.getShape(), elementType);
+    addConversion([](TensorType type) -> Type {
+      if (type.getElementType().isa<TF::StringType>()) {
+        auto elementType = TFStrings::StringType::get(type.getContext());
+        return RankedTensorType::get(type.getShape(), elementType);
```

Comment:
can this trigger an assert for unranked tensors?

Reply:
maybe make this conversion take RankedTensorType instead of just TensorType

Reply:
Switched to RankedTensorType. Long term I should add UnrankedTensorType too.